design/deploy-steps.md

```diff
@@ -0,0 +1,153 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# Customizable deployment procedure
+
+## Status
+
+implementable
+
+## Summary
+
+This design proposes customizing the deployment procedure by providing a high
+level abstraction around Ironic [deploy steps][deploy-steps], as well as
+allowing per-host deployment (IPA) images.
+
+[deploy-steps]: https://docs.openstack.org/ironic/latest/admin/node-deployment.html#deploy-steps
+
+## Motivation
+
+By default the Ironic agent downloads a provided image via HTTP(s), converts it
+(if needed) and writes directly to the disk. This procedure works for many
+distributions, but may not suit some cases well. A prominent example is CoreOS,
+which has its own installer (`coreos-installer`).
+
+### Goals
+
+- A way to replace the default deploy procedure with a custom action.
+- A way to provide a link to the Ironic agent image per host.
+
+### Non-Goals
+
+- Expose the full functionality of Ironic deploy steps.
+- Add a way to execute custom actions during deployment.
+- Building custom agent images.
+
+## Proposal
+
+### User Story
+
+As a CoreOS user I want to use `coreos-installer` to write the CoreOS image
+without losing any functionality that Ironic offers (unlike the live ISO
+workflow).
+
+## Design Details
+
+Additions to `BareMetalHost`:
+
+- New structure field `AgentImage` with the only field `URL`. It will accept
```

Comment:
Instead of a new field, can we instead use the existing `image` alongside `image.format` to mimic this field's behavior? Perhaps we can even add a sub-field under `image` to reflect that this is an custom iso?

Reply:
I'm very explicitly avoiding to mix the Image with the Agent image. They're not the same thing in a normal situation, they don't even have to be based on the same technology (as it's the case now).